### PR TITLE
feat(control): Mobile Ads (Banner and Interstitial)

### LINKED ISSUE #286: Google Mobile and Web Ads integration
Hi Flet team,

This is an extraordinarily beautiful repo for python devs :) 

I am just curious do you have any support now or future for Admob?

### PR DESCRIPTION
**Pull request by ndonkoHenri**
Closes #286 

## Test Code:
```py
import flet as ft
import flet.ads as ads


def main(page: ft.Page):
    page.horizontal_alignment = ft.CrossAxisAlignment.CENTER

    id_interstitial = (
        "ca-app-pub-3940256099942544/1033173712"
        if page.platform == ft.PagePlatform.ANDROID
        else "ca-app-pub-3940256099942544/4411468910"
    )

    id_banner = (
        "ca-app-pub-3940256099942544/6300978111"
        if page.platform == ft.PagePlatform.ANDROID
        else "ca-app-pub-3940256099942544/2934735716"
    )

    def handle_interstitial_close(e):
        nonlocal iad
        print("InterstitialAd closed")
        page.overlay.remove(e.control)
        page.overlay.append(iad := get_new_interstitial_ad())
        page.update()

    def get_new_interstitial_ad():
        return ads.InterstitialAd(
            unit_id=id_interstitial,
            on_load=lambda e: print("InterstitialAd loaded"),
            on_error=lambda e: print("InterstitialAd error", e.data),
            on_open=lambda e: print("InterstitialAd opened"),
            on_close=handle_interstitial_close,
            on_impression=lambda e: print("InterstitialAd impression"),
            on_click=lambda e: print("InterstitialAd clicked"),
        )

    def display_new_banner_ad():
        page.add(
            ft.Container(
                content=ads.BannerAd(
                    unit_id=id_banner,
                    on_click=lambda e: print("BannerAd clicked"),
                    on_load=lambda e: print("BannerAd loaded"),
                    on_error=lambda e: print("BannerAd error", e.data),
                    on_open=lambda e: print("BannerAd opened"),
                    on_close=lambda e: print("BannerAd closed"),
                    on_impression=lambda e: print("BannerAd impression"),
                    on_will_dismiss=lambda e: print("BannerAd will dismiss"),
                ),
                width=320,
                height=50,
                bgcolor=ft.colors.TRANSPARENT,
            )
        )

    page.overlay.append(iad := get_new_interstitial_ad())
    page.appbar = ft.AppBar(
        adaptive=True,
        title=ft.Text("Mobile Ads Playground"),
        bgcolor=ft.colors.LIGHT_BLUE_300,
    )
    page.add(
        ft.OutlinedButton("Show InterstitialAd", on_click=lambda e: iad.show()),
        ft.OutlinedButton("Show BannerAd", on_click=lambda e: display_new_banner_ad()),
    )


ft.app(main)

```

## Summary by Sourcery

This pull request adds support for mobile ads (Banner and Interstitial) to the Flet framework. It includes new classes for managing these ads, updates to the main Dart file for initialization, and modifications to the pubspec.yaml to include the new package. Initial documentation for the new controls is also provided.

- **New Features**:
    - Introduced support for mobile ads, including Banner and Interstitial ads, in the Flet framework.
    - Added new classes for handling mobile ads: `BannerAd`, `InterstitialAd`, and `NativeAd` with associated event handlers for load, error, open, close, impression, click, and will dismiss events.
- **Enhancements**:
    - Updated the main Dart file to initialize the new mobile ads package and ensure it is properly set up.
    - Modified the pubspec.yaml to include the new `flet_mobile_ads` package.
- **Documentation**:
    - Added initial documentation for the new mobile ads controls in the `README.md` file of the `flet_mobile_ads` package.

### CODE REVIEW

**Comment by zox47** [comment]
Can you add (Rewarded interstitial - Interstitial Video - App Open)

**Comment by ndonkoHenri** [comment]
Waiting on https://github.com/flet-dev/flet/issues/3345 to finalize
(the unit ID for the ads is passed as property, but however needs to be passed to iOS .plist and Android .xml)

**Comment by zox47** [comment]
can you add some documentation for native ads

**Comment by zox47** [comment]
.

> Waiting on #3345 to finalize (the unit ID for the ads is passed as property, but however needs to be added to iOS .plist and Android .xml equally)

please can you add native ads in documentation?

**Comment by ndonkoHenri** [comment]
I faced some issues while trying to add Native ads (forgot which one though).
It's files are present in this PR, but the Control can't be accessed nor used as its not fully ready-for-use.

For a start, we will have just Banner and Interstitial ads. Native might come up later on...

File: sdk/python/packages/flet-core/src/flet_core/__init__.py
Comment (FeodorFitsner) [inline sdk/python/packages/flet-core/src/flet_core/__init__.py:1]: `flet_core` has gone.

**Review by FeodorFitsner — APPROVED** [review]

**Comment by ndonkoHenri** [comment]
@sourcery-ai review

**Comment by joesandiroz2** [comment]
i test baner you ad code . and why error in app

"unknow control : banner_ad "

i test in android

**Comment by ndonkoHenri** [comment]
Using the Flet mobile app? Or you packaged as apk?

**Comment by joesandiroz2** [comment]
> Using the Flet mobile app? Or you packaged as apk?

Flet mobile . Yeah it shouldn't work . But is there a way to debug directly into android . Like . Flutter run android . That would be useful. When the development

**Pull request merged**